examples/argv.js

```diff
@@ -0,0 +1,100 @@
+"use strict";
+
+// Run me with Node to see my output!
+// $ node examples/argv.js foo -x
+// $ node examples/argv.js -x=1 bar
+// $ node examples/argv.js --some --thing=x
+
+let util = require("util");
+let P = require("..");
+
+let CLI = P.createLanguage({
+  expression: function(r) {
+    // whitespace-separated words, strings and options
+    return P.alt(r.word, r.string, r.option)
+      .sepBy(P.whitespace)
+      .trim(P.optWhitespace);
+  },
+
+  option: function(r) {
+    // one of possible quotes, then sequence of anything except that quote (unless escaped), then the same quote
+    return P.seq(
+      P.alt(P.string("-").then(P.regex(/[a-z]/)), P.string("--").then(r.word)),
+      P.alt(P.string("=").then(r.word), P.of(true))
+    );
+  },
+
+  word: function() {
+    // 1 char of anything except forbidden symbols and dash, then 0+ chars of anything except forbidden symbols
+    return P.regex(/[^-=\s"'][^=\s"']*/);
+  },
+
+  string: function() {
+    // one of possible quotes, then sequence of anything except that quote (unless escaped), then the same quote
+    return P.oneOf(`"'`).chain(function(q) {
+      return P.alt(
+        P.regex(new RegExp(`[^\\\\${q}]+`)),
```

Comment:
this regexp is pretty confusing to me. i think you might be able to rewrite this using some of the other parsimmon constructors?

you could also just write two separate string rules and then use alt() to combine them

Reply:
Yes, double escaping is going on here, not the cleanest code in the history :)
I agree with both points – will try to rewrite this as you suggested.